### PR TITLE
Remove mercurial support

### PR DESCRIPTION
**Pull request by AlanCoding**
##### SUMMARY
Mercurial SCM was deprecated https://github.com/ansible/awx/issues/7932 and is now being removed.

##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME
 - API

##### AWX VERSION
```
15.0.1
```

marked for next release

##### ADDITIONAL INFORMATION
There will be no support for mercurial moving towards execution environments.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,9 @@
 
 This is a list of high-level changes for each release of AWX. A full list of commits can be found at `https://github.com/ansible/awx/releases/tag/<version>`.
 
+## 16.0.0 (TBD)
+- Removed support for Mercurial-based project updates - https://github.com/ansible/tower/issues/4727
```
Comment (ryanpetrello) [inline CHANGELOG.md:6]: Might want to hold off and rebase my PR: https://github.com/ansible/awx/pull/8723

**Review by jakemcdermott — APPROVED** [review]

**Comment by AlanCoding** [comment]
It's hard to test migrations, so I ran through the manual process and tried to document as well as possible. I have done this twice.

Starting state documentation:

```
In [1]: (ProjectUpdateEvent.objects.filter(project_update__project__scm_type='hg').count(), ProjectUpdate.objects.
   ...: filter(project__scm_type='hg').count(), Project.objects.filter(scm_type='hg').count())
Out[1]: (170, 9, 9)
```

Now I check out the branch and run this migration:

```
bash-4.4$ awx-manage migrate main
Operations to perform:
  Apply all migrations: main
Running migrations:
2020-12-01 20:59:00,486 WARNING  awx.main.migrations Removed 9 mercurial projects, deprecation period ended
  Applying main.0123_drop_hg_support... OK
```

Confirm that the database is in the desired state:

```
In [1]: (ProjectUpdateEvent.objects.filter(project_update__project__scm_type='hg').count(), ProjectUpdate.objects.
   ...: filter(project__scm_type='hg').count(), Project.objects.filter(scm_type='hg').count())
Out[1]: (0, 0, 0)
```

Perfect!

With this, I'm going to say that I'm done with this work.

File: CHANGELOG.md (hunk 1 above)
Comment (ryanpetrello) [reply CHANGELOG.md:6]: Actually, you go first and I'll deal w/ the conflict.

File: awx/settings/local_settings.py.docker_compose [hunk 2]
```diff
@@ -175,13 +175,6 @@ TEST_GIT_PUBLIC_HTTPS = 'https://github.com/ansible/ansible.github.com.git'
 TEST_GIT_PRIVATE_HTTPS = 'https://github.com/ansible/product-docs.git'
 TEST_GIT_PRIVATE_SSH = 'git@github.com:ansible/product-docs.git'
 
-TEST_HG_USERNAME = ''
-TEST_HG_PASSWORD = ''
-TEST_HG_KEY_DATA = TEST_SSH_KEY_DATA
-TEST_HG_PUBLIC_HTTPS = 'https://bitbucket.org/cchurch/django-hotrunner'
```
Comment (ryanpetrello) [inline awx/settings/local_settings.py.docker_compose:181]: image


lol

**Review by ryanpetrello — APPROVED** [review]
:shipit:

**Pull request merged**